.gitignore

```diff
@@ -1,5 +1,13 @@
-build*/*
+build-*/*
+build/*
 external/*
 /.project
 .DS_Store
-_make_installer/*
\ No newline at end of file
+_make_installer/*
+cmake_install.cmake
+.vs/
+out/
+*.depend*
+*.vcxproj*
+libautoscoper/CMakeFiles/*
+autoscoper/CMakeFiles/*
```

Comment:
To systematically prevent in source build, I suggest to look into including module like this one: https://github.com/Slicer/Slicer/blob/main/CMake/PreventInSourceBuilds.cmake